### PR TITLE
Instant message

### PR DESCRIPTION
**Pull request by zycbobby**
Not finished:
- Unread message
- show team members
- Infinite scroll
- Message detail ( How a message show)
- Load default channel when come in

### CODE REVIEW

File: app/stores/UserStore.js [hunk 1]
```diff
@@ -146,10 +148,10 @@ let UserStore = assign({}, BaseStore, {
         return _username_users[name];
     },
     dispatcherIndex: AppDispatcher.register(function (payload) {
-        let data = payload.data;
+        let action = payload.action;
```
Comment (LeonLiuY) [inline app/stores/UserStore.js]: We can remove this.

File: app/stores/UserStore.js [hunk 2]
```diff
@@ -2,12 +2,13 @@
 import AppDispatcher from '../dispatchers/AppDispatcher';
 import Constants from '../constants/AppConstants';
 import BaseStore from './BaseStore';
+import LoginStore from './LoginStore';
```
Comment (LeonLiuY) [inline app/stores/UserStore.js]: We can remove this

File: config/app.jsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ $.ajaxSetup({
     },
     processData: false
 });
-require('jquery-mockjax');
+// require('jquery-mockjax');
```
Comment (LeonLiuY) [inline config/app.jsx]: Uncomment this

File: app/stores/SocketStore.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        let self = this;
+        _socket.removeAllListeners();
+        window.onbeforeunload = function (e) {
+            _socket.emit('user:disconnect', {userId: LoginStore.getUser().id});
+            _socket.disconnect();
+            _socket = null;
+        };
+
+        _socket.on('message:send', function (message) {
+            const MessageStore = require('./MessageStore');
```
Comment (zycbobby) [inline app/stores/SocketStore.js:39]: https://coderwall.com/p/myzvmg/circular-dependencies-in-node-js

**Pull request merged**